scripts/retrieve_databundle_light.py

```diff
@@ -354,6 +356,85 @@ def download_and_unzip_direct(config, rootpath, hot_run=True, disable_progress=F
+    disable_progress=False)
+
+    Function to download and unzip the data for hydrobasins from HydroBASINS database
+    available via https://www.hydrosheds.org/products/hydrobasins
+
+    We are using data from the HydroSHEDS version 1 database
+    which is © World Wildlife Fund, Inc. (2006-2022) and has been used herein under license.
+    WWF has not evaluated our data pipeline and therefore gives no warranty regarding its
+    accuracy, completeness, currency or suitability for any particular purpose.
+    Portions of the HydroSHEDS v1 database incorporate data which are the intellectual property
+    rights of © USGS (2006-2008), NASA (2000-2005), ESRI (1992-1998), CIAT (2004-2006),
+    UNEP-WCMC (1993), WWF (2004), Commonwealth of Australia (2007), and Her Royal Majesty
+    and the British Crown and are used under license. The HydroSHEDS v1 database and
+    more information are available at https://www.hydrosheds.org.
+
+    Inputs
+    ------
+    config : Dict
+        Configuration data for the category to download
+    rootpath : str
+        Absolute path of the repository
+    hot_run : Bool (default True)
+        When true the data are downloaded
+        When false, the workflow is run without downloading and unzipping
+    disable_progress : Bool (default False)
+        When true the progress bar to download data is disabled
+
+    Outputs
+    -------
+    True when download is successful, False otherwise
+    """
+    resource = config["category"]
+    url_templ = config["urls"]["hydrobasins"]
+    suffix_list = config["urls"]["suffixes"]
+
+    level_code = snakemake.config["renewable"]["hydro"]["resource"]["hydrobasins_level"]
+    if level_code <= 9:
+        level_code = "0" + str(level_code)
+    else:
+        level_code = str(level_code)
+
+    for rg in suffix_list:
+        # lev01_v1c.zip
+        url = url_templ + "hybas_" + rg + "_lev" + level_code + "_v1c.zip"
+        file_path = os.path.join(config["destination"], os.path.basename(url))
+        if hot_run:
+            if os.path.exists(file_path):
+                os.remove(file_path)
+
+            try:
+                logger.info(
+                    f"Downloading resource '{resource}' for hydrobasins in '{rg}' from cloud '{url}'."
+                )
+                progress_retrieve(
+                    url,
+                    file_path,
+                    headers=[("User-agent", "Mozilla/5.0")],
+                    disable_progress=disable_progress,
+                )
+
+                with ZipFile(file_path, "r") as zipfile:
+                    zipfile.extractall(config["destination"])
+
+                os.remove(file_path)
+                logger.info(f"Downloaded resource '{resource}' from cloud '{url}'.")
+            except:
+                logger.warning(
+                    f"Failed download resource '{resource}' from cloud '{url}'."
+                )
+                return False
```

Comment:
This clearly resembles the download_direct approach.
Rather than duplicating the code, do you think it is possible to reuse the other function and save space?
That may be possible by creating an ad-hoc config dictionary probably with unzip being true, the headers and the appropriate updated paths, but let me know your thoughts :)

Reply:
Agree that it would be nice to avoid duplication which clearly presents now. Do you mean that we can read all the needed elements from `bundle.config`, even if the structure on the request differ? To me, the required data structure is not yet evident, but happy to build it.

Currently, I see the following differences `download_hydrobasins` from `download_direct`:
1) a loop is needed to go through all the suffixes;
2) header should be specified;
3) HydroBasins license text is added (probably, we can move it to the docstring, for example).

To me, the first point currently looks the most tricky one. Do you have any ideas how to deal with that?

Reply:
My suggestion was to propose calling
`download_and_unzip_direct(..., ..., hot_run=hot_run, disable_progress=disable_progress)`
within this for loop, rather than copying the code again, but we can go for the duplication if it is easier